src/agent/library/skills.js

```diff
@@ -491,13 +491,28 @@ export async function placeBlock(bot, blockType, x, y, z) {
      * await skills.placeBlock(bot, "oak_log", position.x + 1, position.y - 1, position.x);
      **/
     console.log('placing block...')
+    let block = bot.inventory.items().find(item => item.name === blockType);
+    if (!block) {
+        log(bot, `Don't have any ${blockType} to place.`);
+        return false;
+    }
+
     const target_dest = new Vec3(Math.floor(x), Math.floor(y), Math.floor(z));
-    const empty_blocks = ['air', 'water', 'lava', 'grass', 'tall_grass', 'snow', 'dead_bush', 'fern'];
     const targetBlock = bot.blockAt(target_dest);
-    if (!empty_blocks.includes(targetBlock.name)) {
-        log(bot, `Cannot place block at ${targetBlock.position} because ${targetBlock.name} is in the way.`);
+    if (targetBlock.name === blockType) {
+        log(bot, `${blockType} already at ${targetBlock.position}.`);
```

Comment:
Should this log (or a follow up log) mention that this block was removed so that the bot doesn't think the task failed?

Reply:
well here it doesn't remove/place anything, since the target block is already the block to be placed. if it does remove something, breakBlockAt should log that it removed it. 

This returns false since it didn't place anything, but I think usually the return value is ignored anyway

Reply:
Ok. I didn't notice that breakBlockAt creates a log. That resolves my concern.